sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FluxByteBufferContent.java

```diff
@@ -46,6 +46,9 @@ public FluxByteBufferContent(Flux<ByteBuffer> content, Long length) {
 
     @Override
     public Long getLength() {
+        if (bytes.get() != null) {
```

Comment:
This concerns me as it'll read the `Flux<ByteBuffer>` into memory eagerly in all cases where the `HttpRequest` is based on `Flux<ByteBuffer>`, potentially running into OutOfMemoryErrors in SDKs such as Storage.

Reply:
That's not how this works. This just checks if `toBytes()` has been called. If it has (and cached the result) then content is known. Otherwise it isn't.

There's even test for that in this PR.
https://github.com/kasobol-msft/azure-sdk-for-java/blob/8e3270a2c9d796fd4da4498db18a79f7dec3c7b7/sdk/core/azure-core/src/test/java/com/azure/core/http/HttpRequestTests.java#L169

Reply:
also here https://github.com/kasobol-msft/azure-sdk-for-java/blob/d0a4dec868eca9cdf3c1633508f7d79a083a4a33/sdk/core/azure-core/src/test/java/com/azure/core/util/BinaryDataTest.java#L241-L254

Reply:
Right, got this confused with calling `toBytes()`